src/vm/invokeutil.cpp

```diff
@@ -649,9 +649,14 @@ void InvokeUtil::ValidField(TypeHandle th, OBJECTREF* value)
         COMPlusThrow(kArgumentException,W("Arg_ObjObj"));
 }
 
-// InternalCreateObject
-// This routine will create the specified object from the value
-OBJECTREF InvokeUtil::CreateObject(TypeHandle th, void * pValue) {
+//
+// CreateObjectAfterInvoke
+// This routine will create the specified object from the value returned by the Invoke target. 
+//
+// This does not handle the ELEMENT_TYPE_VALUETYPE case. There is no way for this function to handle that case in a GC-safe way.
```

Comment:
> There is no way for this function to handle that case in a GC-safe way.

I do not think that this is valid statement. It can be handled bu protecting `pValue` using `GCPROTECT_BEGININTERIOR`.

Reply:
Is that also safe when `pValue` is a completely unmanaged pointer (i.e. into the stack?)

Reply:
(I think it is fine for this function to not handle `ELEMENT_TYPE_VALUETYPE`.)

Reply:
Ok, we'll delete the phrase from the comment, then.

Reply:
> Is that also safe when pValue is a completely unmanaged pointer (i.e. into the stack?)

Yes.